tokio/src/runtime/metrics/batch.rs

```diff
@@ -39,87 +52,107 @@ pub(crate) struct MetricsBatch {
+    struct PollTimer {
+        /// Histogram of poll counts within each band.
+        poll_counts: HistogramBatch,
 
-    /// Instant when the most recent task started polling.
-    poll_started_at: Instant,
+        /// Instant when the most recent task started polling.
+        poll_started_at: Instant,
+    }
 }
 
 impl MetricsBatch {
-    pub(crate) fn new(worker_metrics: &WorkerMetrics) -> MetricsBatch {
+    pub(crate) fn new(_worker_metrics: &WorkerMetrics) -> MetricsBatch {
         let now = Instant::now();
+        #[cfg(not(tokio_unstable))]
+        {
+            MetricsBatch {
+                busy_duration_total: 0,
+                processing_scheduled_tasks_started_at: now,
+            }
+        }
 
-        MetricsBatch {
-            park_count: 0,
-            park_unpark_count: 0,
-            noop_count: 0,
-            steal_count: 0,
-            steal_operations: 0,
-            poll_count: 0,
-            poll_count_on_last_park: 0,
-            local_schedule_count: 0,
-            overflow_count: 0,
-            busy_duration_total: 0,
-            processing_scheduled_tasks_started_at: now,
-            poll_timer: worker_metrics
-                .poll_count_histogram
-                .as_ref()
-                .map(|worker_poll_counts| PollTimer {
-                    poll_counts: HistogramBatch::from_histogram(worker_poll_counts),
-                    poll_started_at: now,
-                }),
+        #[cfg(tokio_unstable)]
+        {
+            MetricsBatch {
+                park_count: 0,
+                park_unpark_count: 0,
+                noop_count: 0,
+                steal_count: 0,
+                steal_operations: 0,
+                poll_count: 0,
+                poll_count_on_last_park: 0,
+                local_schedule_count: 0,
+                overflow_count: 0,
+                busy_duration_total: 0,
+                processing_scheduled_tasks_started_at: now,
+                poll_timer: _worker_metrics.poll_count_histogram.as_ref().map(
+                    |worker_poll_counts| PollTimer {
+                        poll_counts: HistogramBatch::from_histogram(worker_poll_counts),
+                        poll_started_at: now,
+                    },
+                ),
+            }
         }
     }
 
-    pub(crate) fn submit(&mut self, worker: &WorkerMetrics, mean_poll_time: u64) {
-        worker.mean_poll_time.store(mean_poll_time, Relaxed);
-        worker.park_count.store(self.park_count, Relaxed);
-        worker
-            .park_unpark_count
-            .store(self.park_unpark_count, Relaxed);
-        worker.noop_count.store(self.noop_count, Relaxed);
-        worker.steal_count.store(self.steal_count, Relaxed);
-        worker
-            .steal_operations
-            .store(self.steal_operations, Relaxed);
-        worker.poll_count.store(self.poll_count, Relaxed);
-
+    pub(crate) fn submit(&mut self, worker: &WorkerMetrics, _mean_poll_time: u64) {
```

Comment:
As per the comment on `unparked`, let's follow the convension elsewhere in the Tokio codebase where we need different implementations for stablized vs. unstablized functionality. Here that would mean the following:

```rust
pub(crate) fn submit(&mut self, worker: &WorkerMetrics, mean_poll_time: u64) {
    worker
        .busy_duration_total
        .store(self.busy_duration_total, Relaxed);

    self.submit_unstable(worker, mean_poll_time);
}

cfg_not_unstable_metrics! {
    #[inline(always)]
    fn submit_unstable(&mut self, _worker: &WorkerMetrics, _mean_poll_time: u64) {}
}

cfg_unstable_metrics! {
    #[inline(always)]
    fn submit_unstable(&mut self, worker: &WorkerMetrics, mean_poll_time: u64){
        worker.mean_poll_time.store(_mean_poll_time, Relaxed);
        worker.park_count.store(self.park_count, Relaxed);
        worker
            .park_unpark_count
            .store(self.park_unpark_count, Relaxed);
        worker.noop_count.store(self.noop_count, Relaxed);
        worker.steal_count.store(self.steal_count, Relaxed);
        worker
            .steal_operations
            .store(self.steal_operations, Relaxed);
        worker.poll_count.store(self.poll_count, Relaxed);

        worker
            .local_schedule_count
            .store(self.local_schedule_count, Relaxed);
        worker.overflow_count.store(self.overflow_count, Relaxed);

        if let Some(poll_timer) = &self.poll_timer {
            let dst = worker.poll_count_histogram.as_ref().unwrap();
            poll_timer.poll_counts.submit(dst);
        }
    }
}
```

Use the same `cfg_unstable_metrics` and `cfg_not_unstable_metrics` blocks as for the other functions that need them (so there is only one of each in `impl MetricsBatch`.